### PR TITLE
Convert both namespace and limitrange examples to v1beta3

### PR DESCRIPTION
**Pull request by dchen1107**
changes: 1) apiVersion 2) replace id with name

### CODE REVIEW

**Comment by dchen1107** [comment]
#3671

File: examples/limitrange/v1beta3/invalid-pod.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    image: kubernetes/serve_hostname
+    imagePullPolicy: IfNotPresent
+    name: kubernetes-serve-hostname
+    resources: {}
+    terminationMessagePath: /dev/termination-log
+  dnsPolicy: ClusterFirst
+  restartPolicy:
+    always: {}
+  volumes: null
+status: {}
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/invalid-pod.json]: Remove unnecessary fields, like status.

**Comment by dchen1107** [comment]
Manually redo the conversion now since the tool not just generates default fields, but also convert json to yaml.

File: examples/kubernetes-namespaces/v1beta3/README.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
+## Kubernetes Namespaces
+
+Kubernetes Namespaces help different projects, teams, or customers to share a Kubernetes cluster.
+
+It does this by providing the following:
+
+1. A scope for [Names](../../docs/identifiers.md).
```
Comment (bgrant0607) [inline examples/kubernetes-namespaces/v1beta3/README.md]: I think you need one more level of `../`

File: examples/kubernetes-namespaces/v1beta3/README.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+are relaxed to enable agile development.
+
+The operations team would like to maintain a space in the cluster where they can enforce strict procedures on who can or cannot manipulate the set of
+pods, services, and replication controllers that run the production site.
+
+One pattern this organization could follow is to partition the Kubernetes cluster into two namespaces: development and production.
+
+Let's create two new namespaces to hold our work.
+
+Use the file `examples/kubernetes-namespaces/namespace-dev.json` which describes a development namespace:
```
Comment (bgrant0607) [inline examples/kubernetes-namespaces/v1beta3/README.md]: v1beta3

File: examples/kubernetes-namespaces/v1beta3/README.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  "labels": {
+    "name": "development"
+  },
+}
+```
+
+Create the development namespace using kubectl.
+
+```shell
+$ cluster/kubectl.sh create -f examples/kubernetes-namespaces/namespace-dev.json
```
Comment (bgrant0607) [inline examples/kubernetes-namespaces/v1beta3/README.md]: v1beta3

File: examples/kubernetes-namespaces/v1beta3/README.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+```
+
+And then lets create the production namespace using kubectl.
+
+```shell
+$ cluster/kubectl.sh create -f examples/kubernetes-namespaces/namespace-prod.json
```
Comment (bgrant0607) [inline examples/kubernetes-namespaces/v1beta3/README.md]: All these paths are wrong.

File: examples/limitrange/v1beta3/valid-pod.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  "kind": "Pod",
+  "apiVersion":"v1beta3",
+  "labels": {
+    "name": "valid-pod"
+  },
+  "spec": {
+    "containers": [{
+      "name": "kubernetes-serve-hostname",
+      "image": "kubernetes/serve_hostname",
+      "cpu": 1000,
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/valid-pod.json]: Please use the new resources map.

**Comment by dchen1107** [comment]
Ok, done it. PATL?

**Comment by derekwaynecarr** [comment]
I would like to try the LimitRange changes tomorrow to be sure that units are still right.  I thought units changed between beta1 and beta3.

File: examples/limitrange/v1beta3/valid-pod.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  "labels": {
+    "name": "valid-pod"
+  },
+  "spec": {
+    "containers": [{
+      "name": "kubernetes-serve-hostname",
+      "image": "kubernetes/serve_hostname",
+        "resources": {
+          "limits": {
+            "cpu": 1000,
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/valid-pod.json]: Should be "1" (must be a string and is in cores, not millicores).

File: examples/limitrange/v1beta3/valid-pod.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+            "memory": 1048576,
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/valid-pod.json]: Should be a string: "1Mi"

File: examples/limitrange/v1beta3/limit-range.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+      }
+    },
+    {
+      "type": "Container",
+      "max": {
+        "memory": "1073741824",
+        "cpu": "2",
+      },
+      "min": {
+        "memory": "1048576",
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/limit-range.json]: "1Mi"

File: examples/limitrange/v1beta3/limit-range.json [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        "cpu": "0.25"
+      }
+    },
+    {
+      "type": "Container",
+      "max": {
+        "memory": "1073741824",
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/limit-range.json]: "1Gi"

File: examples/limitrange/v1beta3/limit-range.json [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    "limits": [
+    {
+      "type": "Pod",
+      "max": {
+        "memory": "1073741824",
+        "cpu": "2",
+      },
+      "min": {
+        "memory": "1048576",
+        "cpu": "0.25"
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/limit-range.json]: "250m"

File: examples/limitrange/v1beta3/limit-range.json [hunk 12]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "limits",
+  "kind": "LimitRange",
+  "apiVersion": "v1beta3",
+  "spec": {
+    "limits": [
+    {
+      "type": "Pod",
+      "max": {
+        "memory": "1073741824",
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/limit-range.json]: "1Gi"

**Comment by dchen1107** [comment]
Fixed units issue, and run all example against v1beta3. Everything looks fine except invalid_pod is not rejected, and I believe because we don't enable mission control by default yet. 

$ cluster/kubectl.sh --api-version=v1beta3  create -f examples/limitrange/v1beta3/valid-pod.json
current-context: "golden-system-455_kubernetes"

$ cluster/kubectl.sh --api-version=v1beta3  get  pods valid-pod
current-context: "golden-system-455_kubernetes"
Running: cluster/../cluster/gce/../../_output/dockerized/bin/linux/amd64/kubectl --api-version=v1beta3 get pods valid-pod
POD                 IP                  CONTAINER(S)                IMAGE(S)                    HOST                                                                 LABELS              STATUS              CREATED
valid-pod           10.244.2.117        kubernetes-serve-hostname   kubernetes/serve_hostname   kubernetes-minion-gs3d.c.golden-system-455.internal/104.197.14.122   <none>              Pending             19 seconds

File: examples/limitrange/v1beta3/valid-pod.json [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+  "labels": {
+    "name": "valid-pod"
+  },
+  "spec": {
+    "containers": [{
+      "name": "kubernetes-serve-hostname",
+      "image": "kubernetes/serve_hostname",
+        "resources": {
+          "limits": {
+            "cpu": 1,
```
Comment (bgrant0607) [inline examples/limitrange/v1beta3/valid-pod.json]: Must be strings!!!

**Comment by bgrant0607** [comment]
Reassigning to @derekwaynecarr for final review.

**Comment by dchen1107** [comment]
Addressed all comments. Surprisingly when I tested it against v1beta3 apiserver, all errors are not detected by validation code. :-( PTAL

**Comment by derekwaynecarr** [comment]
It would be rejected when it's enabled which I will do for all salt providers hopefully tomorrow.

**Comment by derekwaynecarr** [comment]
I tested the limit range scenario on a server with the admission control checks in place and invalid-pod is properly rejected and valid-pod is properly accepted.

LGTM.

**Comment by dchen1107** [comment]
@derekwaynecarr Did you ever think about enable that module by default? It won't have any impact if there is no limits object created anyway, right? :-)

**Comment by derekwaynecarr** [comment]
@dchen1107 - I am literally doing that right now for all the salt based providers in a PR I am testing on GCE now ;-)

When #4749 merges, it has no substantive perf impact to not always have on across all providers.

**Comment by dchen1107** [comment]
Good to know that. Thanks!

**Pull request merged**